### PR TITLE
Binary for Debian 9 Arm64

### LINKED ISSUE #121: Support for Arm64
Hello, 

Thanks for your work.
Do you have plan to support arm64 ?  
I am using wicked_pdf in a debian docker on a M1 macbook) 

The fall back for 
```
os =>  debian_9
RbConfig::CONFIG['host_cpu’]  => arch64
```
is the suffix`debian_9_i386` which throw the error: 
```Command Error: qemu-i386: Could not open '/lib/ld-linux.so.2': No such file or directory```

For those in the same situation a **workaround**  is to install the arm64 build for your distro from
https://wkhtmltopdf.org/downloads.html

install it with
```sh
apt install /path/to/debfile.deb
```

And configure wiked_pdf with 
```ruby
WickedPdf.config = {
  exe_path: '/usr/local/bin/wkhtmltopdf',
}
```

### PR DESCRIPTION
**Pull request by pedrofurtado**
Closes #121 

@unixmonkey 🤝 🤓 

PS: Binary extracted and gzipped from official site of Wkhtmltopdf 🥇

### CODE REVIEW

**Comment by pedrofurtado** [comment]
@stanBienaives Can you help us, testing this binary in your debian 9 arm64, to confirm if it is working well? 👍  I don't have a mac M1 😆

**Comment by stanBienaives** [comment]
Yes I'll do that tmrw morning

**Comment by stanBienaives** [comment]
👋 @pedrofurtado 

I tested the binary with the container ruby:2.6.1  (arm64 build) on my M1 machine and it  worked fine with the command
`.bin//wkhtmltopdf www.google.fr test.pdf`


Do you need more info ? 
I'll be happy to test other distro on arm64 in the future if needed.

**Comment by viktorianer** [comment]
I tested it in my Gemfile with `gem "wkhtmltopdf-binary", git: "https://github.com/pedrofurtado/wkhtmltopdf_binary_gem", ref: "fcab0b766054bc41b2aeac71ff2d687d951bae74"` with a ruby:3.0.2-slim-buster (arm64 build) container on M1 Pro, and it fails with the error:`qemu-i386: Could not open '/lib/ld-linux.so.2': No such file or directory`.

In the container, the command `/usr/local/bundle/bin/wkhtmltopdf -v` gives me the error:

```bash
/usr/local/bundle/bin/wkhtmltopdf -v
/usr/local/lib/ruby/3.0.0/rubygems.rb:281:in `find_spec_for_exe': can't find gem wkhtmltopdf-binary (>= 0.a) with executable wkhtmltopdf (Gem::GemNotFoundException)
	from /usr/local/lib/ruby/3.0.0/rubygems.rb:300:in `activate_bin_path'
	from /usr/local/bundle/bin/wkhtmltopdf:23:in `<main>'
```

Installing it from [wkhtmltopdf.org](https://wkhtmltopdf.org/downloads.html), as suggested in https://github.com/zakird/wkhtmltopdf_binary_gem/issues/121, works for me perfect!

```Docker
RUN  DEBIAN_FRONTEND=noninteractive apt-get install -yq --no-install-recommends --no-install-suggests \
    ...
    libxrandr2 \
    ... 
    xfonts-75dpi \
    xfonts-base && \
    curl -L -o wkhtmltopdf.deb https://github.com/wkhtmltopdf/packaging/releases/download/0.12.6-1/wkhtmltox_0.12.6-1.buster_$ARCH.deb && \
    dpkg -i ./wkhtmltopdf.deb && \
    rm wkhtmltopdf.deb && \
    ...
```

And then, it works! Notes the libs it depends on (only for slim-buster?)! 

```bash
/usr/local/bin/wkhtmltopdf www.google.fr test.pdf
Loading pages (1/6)
Counting pages (2/6)
Resolving links (4/6)
Loading headers and footers (5/6)
Printing pages (6/6)
Done
```

Also, if I will run test in container as expected!

**Comment by encounter** [comment]
Here's a Dockerfile snippet similar to the above that autodetects architecture and Debian codename, and doesn't require manual dependency installation:

```dockerfile
RUN WK_VERSION=0.12.6-1 \
    && ARCH=$(dpkg --print-architecture) \
    && CODENAME=$(. /etc/os-release; echo $VERSION_CODENAME) \
    && curl -LSfso wkhtmltopdf.deb https://github.com/wkhtmltopdf/packaging/releases/download/$WK_VERSION/wkhtmltox_$WK_VERSION.${CODENAME}_$ARCH.deb \
    && apt-get -q update \
    && apt-get install -qy ./wkhtmltopdf.deb \
    && rm -r wkhtmltopdf.deb /var/lib/apt/lists/*
```

**Comment by pedrofurtado** [comment]
@unixmonkey Hi! 👋 

I think we are to merge ✅

File: bin/wkhtmltopdf [hunk 1]
```diff
@@ -36,13 +39,15 @@ suffix = case RbConfig::CONFIG['host_os']
...
+           architecture ||= 'arm64' if os_based_on_debian_9 && 
+                                       (host_cpu == 'arm64' || host_cpu == 'aarch64' || host_cpu == 'arch64')
 
            os = 'debian_10' if !os_based_on_debian_9 && os.start_with?('debian')
 
            os = 'archlinux' if os.start_with?('arch_') ||
                                os.start_with?('manjaro_')
 
-           architecture = RbConfig::CONFIG['host_cpu'] == 'x86_64' ? 'amd64' : 'i386'
+           architecture ||= host_cpu == 'x86_64' ? 'amd64' : 'i386'
```
Comment (unixmonkey) [inline bin/wkhtmltopdf:50]: @pedrofurtado Do you think we could pull this architecture logic out into something a little more separated?

Maybe a method or variable like this:
```ruby
def architecture
  host_cpu = RbConfig::CONFIG['host_cpu']
  if %w[arm64 aarch64 arch64].include?(host_cpu)
    'arm64'
  else
    host_cpu
  end
end
```

I suppose that could theoretically change the binary chosen for some OSes, so it'd be nice to see if the Dockerfiles catch anything.

**Comment by gaspo53** [comment]
Hey! Can we get this merged? Can't use wickedpdf y arm64 beacuse of the lack of this. Thanks!

**Comment by unixmonkey** [comment]
@gaspo53 Why not? Add It to your `Gemfile` like this:
```ruby
git_source(:github) { |repo| "https://github.com/#{repo}.git" }
gem 'wkhtmltopdf_binary_gem', github: 'pedrofurtado/wkhtmltopdf_binary_gem', ref: 'fcab0b766054bc41b2aeac71ff2d687d951bae74'
```

**Comment by unixmonkey** [comment]
Thank you! Although I didn't merge this specific PR, these Debian binaries are now in version `0.12.6.7` (from [this PR](https://github.com/zakird/wkhtmltopdf_binary_gem/pull/145)). Please check it and make sure it works for you.

**Comment by gaspo53** [comment]
@unixmonkey thanks!

**Pull request closed without merging**